_posts/en/newsletters/2018-10-09-newsletter.md

```diff
@@ -0,0 +1,283 @@
+transaction fee paid by users of Lightning Network payment channels.
+
+*[video][vid scriptless ecdsa], [transcript][tx scriptless ecdsa], possible
+now*
+
+## Discussion wrap-up: the evolution of bitcoin script
+
+*Discussion wrap-up presented by Olaoluwa Osuntokun*
+
+This wrap-up of a two-hour discussion group briefly mentioned a large
+variety of proposed changes to Bitcoin's Script language---far too many
+to mention here even in summary.  However, a few changes were mentioned
+as theoretically possible to accomplish in 2019 if the community is
+willing to adopt them:
+
+- **Schnorr signature scheme:** an opt-in feature providing smaller
+  signatures in all cases, much smaller public key and signature data
+  for cooperative multisigs, and faster validation.  Easier
+  compatibility with scriptless scripts.  See Pieter Wuille's [proposed
+  BIP][schnorr pre-bip].
+
+- **SIGHASH_NOINPUT_UNSAFE:** the ability to create spends without
+  explicitly referencing which output you want to spend.  Allows
+  creating more efficient payment channels using the [Eltoo protocol][]
+  that also makes it easy for each channel to support up to 150
+  participants.  See [BIP118][].
+
+- **OP_CHECKSIGFROMSTACK:** makes it possible to create covenants that
+  restrict what outputs a particular coin can be spent to.  For example,
+  you could require that your bitcoins can only be sent to an address
+  that requires giving you 1,000 blocks (1 week) to evaluate the
+  situation and possibly send them back to a whitelisted address before
+  they can be spent to any non-whitelisted address.  (The presenter
+  notes that some developers are opposed to enabling this for
+  fungibility reasons, although alternative approaches may be more
+  acceptable.)
+
+- **Fixing the time warp bug:** a set of miners controlling a majority of
+  the hash rate can currently manipulate Bitcoin's difficulty adjustment
+  algorithm to allow them to consistently create more than one block
+  every ten minutes even without increasing overall hash rate.  There's
+  at least one simple proposal to reduce the amount of manipulation
+  possible without breaking older software or mining equipment.  See the
+  recent [email thread][bitcoin-dev timewarp] on the Bitcoin-Dev mailing
+  list.
+
+- **Explicit fees:** currently fees in Bitcoin are implied by the difference
+  between the value of the aggregated inputs and the aggregated outputs.
+  However, the transaction could alternatively explicitly commit to the
+  fee and allow one of the outputs to be set to the difference between
+  the value of the aggregated inputs and the explicit fee plus all the
+  other outputs.  This could be useful for rewarding Lightning Network
+  watchtowers that send breach remedy transactions on behalf of offline
+  users, or it could be useful for fee bumping group transactions.
+
+However, the presenter concluded this section of the wrap-up by saying,
```

Comment:
This section should be attributed to a member of the workshop; it wasn't @roasbeef that said this quote.